docs/src/index.md

```diff
@@ -6,4 +6,369 @@ DocTestSetup = :(using ChainRulesCore, ChainRules)
+- **Pullback**
+   -  return by `rrule`
+   - takes output space wobbles, gives input space wibbles
+   - 1 argument per original function return
+   - 1 return per orignal function argument + 1 for the function itself
+
+#### Pushforward/Pullback and Total Derivative/Gradient
+
+The most trivial use of the frule+pushforward is to calculate the [Total Derivative](https://en.wikipedia.org/wiki/Total_derivative):
+```julia
+y, pushforward = frule(f, a, b, c)
+ẏ = pushforward(1, 1, 1, 1)  # for appropriate `1`-like perturbation.
+```
+Then we have that
+`ẏ` is the _total derivative_ of
+`f` at `(a, b, c)`:
+written mathematically as ``df_{(a,b,c)}``
+
+
+Similarly:
+The most trivial use of the rrule+pullback is to calculate the [Gradient](https://en.wikipedia.org/wiki/Gradient):
+```julia
+y, pullback = frule(f, a, b, c)
+∇f  = pushforward(1) # for appropriate `1`-like seed.
+s̄, ā, b̄, c̄ = ∇f
+```
+Then we have that
+`∇f` is the _gradient_ of
+`f` at `(a, b, c)`.
+And we thus have the partial derivative:
+s̄, ā, b̄, c̄.
+(Including the and the self-partial derivative,
+s̄).
+Written mathematically as ``\dfrac{∂f}{∂a}``, ``\dfrac{∂f}{∂b}``, ``\dfrac{∂f}{∂c}``.
+
+### Differentials
+
+The values that come back from pullbacks,
+or pushforwards
+are not always the same type as the input/outputs of the original function.
+They are differentials,
+differency-equivalents.
+A differential might be such a regular type,
+like a Number, or a Matrix,
+or it might be one of the `AbstractDifferential` subtypes.
+
+Differentials support a number of operations.
+Most importantly:
+`+` and `*` which lets them act as mathematically objects.
+And `extern` which converts them into a conventional type.
+
+The most important AbstractDifferentials when getting started are the ones about avoiding work:
+
+ - `Thunk`: this is a deferred computation. A thunk is a [word for a zero argument closure](https://en.wikipedia.org/wiki/Thunk). An computation wrapped in a `@thunk` doesn't get evaluated until `extern` is called on the `Thunk`. More on thunks later.
+ - `One`, `Zero`: There are special representions of `1` and `0`. They do great things around avoiding expanding `Thunks` in multiplication and (for `Zero`) addition.
+
+#### Others: don't worry about them right now
+ - Wirtinger: it is complex. The docs need to be better. [Read the links in this issue](https://github.com/JuliaDiff/ChainRulesCore.jl/issues/40).
+ - Casted: it implements broadcasting mechanics. See [#10](https://github.com/JuliaDiff/ChainRulesCore.jl/issues/10)
+ - InplacableThunk: it is like a Thunk but it can do `store!` and `accumulate!` inplace.
+
+
+ -------------------------------
+## Example of using ChainRules directly.
+
+While ChainRules is largely intended as a backend for Autodiff systems it can be used directly.
+(Infact this can be very useful if you can constraint the code you need to differnetiate to only use thing that have rules defined for.
```

Comment:
```suggestion
(In fact, this can be very useful if you can constraint the code you need to differentiate to only use things that have rules defined for.
```